src/util-streaming-buffer.c

```diff
@@ -0,0 +1,689 @@
+/* Copyright (C) 2015-2016 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+#include "suricata-common.h"
+#include "util-streaming-buffer.h"
+#include "util-unittest.h"
+#include "util-print.h"
+
+/**
+ * \file
+ *
+ * \author Victor Julien <victor@inliniac.net>
+ *
+ *  \brief Streaming Buffer API
+ */
+
```

Comment:
My first thought might be a little silly.  I think "StreamBuffer" flows better.  This buffer isn't streaming itself, but tracks a stream, and its a little less typing in for a utility that I think will also be heavily used.

A little more usage doc as well.  For example, I think I could use this to replace my very simple buffer's I'm using in DNP3 (https://github.com/jasonish/suricata/blob/dnp3/src/app-layer-dnp3.c#L579), but I'm not 100% sure yet without reading all the code.